File: EasyTownHall/Backend/src/main/java/de/othr/easytownhall/models/dtos/workschedule/WorkscheduleDTO.java
package de.othr.easytownhall.models.dtos.workschedule;

import de.othr.easytownhall.models.entities.appointment.TimeSlot;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class WorkscheduleDTO {
    private Long id;

    private Long departmentId;

    private Long serviceworkerId;

    private LocalDate date;

    private List<TimeSlotDto> timeSlots;
}

File: EasyTownHall/Frontend/src/app/models/DTOs/Appointments/WorkscheduleDTO.ts
import { TimeSlotDto } from "./TimeSlotDTO";

export interface WorkscheduleDTO {
    id: number;
    departmentId: number;
    serviceworkerId: number;
    date: string; // ISO format, e.g., '2023-12-25'
    timeSlots: TimeSlotDto[];
  }